addons/portal/static/src/js/portal_share.js

```diff
@@ -0,0 +1,157 @@
+odoo.define('portal.share', function (require) {
+'use strict';
+
+var core = require('web.core');
+var Dialog = require('web.Dialog');
+var SystrayMenu = require('web.SystrayMenu');
+var WebClient = require('web.WebClient');
+var web_settings_dashboard = require('web_settings_dashboard');
+var Widget = require('web.Widget');
+
+var _t = core._t;
+
+var PortalInvitation = web_settings_dashboard.DashboardInvitations.extend({
+    template: 'PortalUserInvitations',
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickInvite: function (ev) {
+        this._createBadges();
+        if (!this.emails.length) {
+            return $.when();
+        }
+        return this._rpc({
+            model: 'res.users',
+            method: 'web_dashboard_create_users',
+            args: [this.emails],
+        });
+    },
+});
+
+var PortalShareDoc = Widget.extend({
+    template: 'PortalSharingIcon',
+    events: {
+        "click": "_onClick",
+    },
+
+    /**
+     * @override
+     */
+    willStart: function () {
+        var self = this;
+        var def = this.getSession().user_has_group('base.group_erp_manager').then(function (has_group){
+            self.use_invite = has_group;
+        });
+        return $.when(def, this._super(this, arguments)).then(function () {
+            core.bus.on('share_action', self, self._toggleShareAction);
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * open user invitation dialog
+     * @private
+     */
+    _openInvitationDialog: function () {
+        this.$content = $("<div>");
+        var invitation = new PortalInvitation(this, {});
+        invitation.appendTo(this.$content);
+        return new Dialog(this, {
+            title: _t('Invite New Users'),
+            size: 'medium',
+            $content: this.$content,
+            buttons: [{
+                text: 'Invite',
+                classes: 'btn-primary',
+                click: function (ev) {
+                    var self = this;
+                    invitation._onClickInvite(ev).then(function (data) {
+                        self.close(true);
+                    });
+                },
+            }, {
+                text: 'Cancel',
+                close: true
+            }]
+        }).open();
+    },
+
+    /**
+     * Update the system tray icon for share document, based on view type
+     * Show Share Icon only if 1) action context has share_icon True 2) form view of portal.mixin inherited object
+     * @private
+     */
+    _toggleShareAction: function (widget) {
+        this._controller = widget;
+        if (widget && widget.viewType === 'form') {
+            if (widget.activeActions && widget.activeActions.share) {
```

Comment:
do action button on every form view for now. We will see with framework team if something is possible to put button next to create one